src/content/structured/get-started/testing-with-jest-and-rtl.mdx

```diff
@@ -0,0 +1,178 @@
+---
+
+import testFiles from "./test-app/index";
+
+export const snippetArray = [
+  {
+    fileName: testFiles.rtl.fileName,
+    disableMoreButton: testFiles.rtl.disableMoreButton,
+    snippets: {
+      short: testFiles.rtl.snippets.short,
+      long: testFiles.rtl.snippets.long,
+    },
+  },
+  {
+    fileName: testFiles.component.fileName,
+    disableMoreButton: testFiles.component.disableMoreButton,
+    snippets: {
+      short: testFiles.component.snippets.short,
+      long: testFiles.component.snippets.long,
+    },
+  },
+  {
+    fileName: testFiles.types.fileName,
+    disableMoreButton: testFiles.types.disableMoreButton,
+    snippets: {
+      short: testFiles.types.snippets.short,
+      long: testFiles.types.snippets.long,
+    },
+  },
+  {
+    fileName: testFiles.constants.fileName,
+    disableMoreButton: testFiles.constants.disableMoreButton,
+    snippets: {
+      short: testFiles.constants.snippets.short,
+      long: testFiles.constants.snippets.long,
+    },
+  },
+  {
+    fileName: testFiles.methods.fileName,
+    disableMoreButton: testFiles.methods.disableMoreButton,
+    snippets: {
+      short: testFiles.methods.snippets.short,
+      long: testFiles.methods.snippets.long,
+    },
+  },
+];
+
+## About Jest
+
+<p>
+  <ic-link
+    target="_blank"
+    href="https://jestjs.io/"
+    rel="noreferer noopener nofollow"
+  >
+    Jest
+  </ic-link>{" "}
+  is a JavaScript testing framework designed to ensure correctness of any
+  JavaScript codebase. It allows you to write tests with an approachable,
+  familiar and feature-rich API that gives you results quickly.
+</p>
+
+<p>
+  These instructions have been written with JavaScript in mind. Jest also
+  supports TypeScript. Further instructions on how to set up Jest for TypeScript
+  can be found on their{" "}
+  <ic-link
+    target="_blank"
+    href="https://jestjs.io/docs/getting-started#using-typescript"
+    rel="noreferer noopener nofollow"
+  >
+    'Getting Started'
+  </ic-link>{" "}
+  page.
+</p>
+
+<p>
+  More information on setting up Jest can be found on{" "}
+  <ic-link
+    target="_blank"
+    href="https://jestjs.io/docs/getting-started"
+    rel="noreferer noopener nofollow"
+  >
+    Jest’s Getting Started page
+  </ic-link>
+  .
+</p>
+
+## Testing with React components
+
+The `@ukic/react` package will need to be transformed before you can use these components in Jest tests.
+
+Add a `transformIgnorePatterns` field with the value `["/node_modules/(?!@ukic/react)"]` to your Jest config.
+
+## About React Testing Library
+
+<p>
+  <ic-link
+    target="_blank"
+    href="https://testing-library.com/docs/react-testing-library/intro"
+    rel="noreferer noopener nofollow"
+  >
+    React Testing Library (RTL)
+  </ic-link>{" "}
+  provides a set of utility functions, which interrogate the DOM tree, in order
+  to build maintainable React tests. The recommended approach with React Testing
```

Comment:
I know this isn't new text but just spotted it now - the abbreviation can be used here instead as it's been defined in the previous line, so "The recommended approach with RTL...".